torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -522,14 +522,42 @@ Tensor clamp_backward(const Tensor & grad, const Tensor &self, const optional<Sc
   }
 }
 
-Tensor mm_mat1_backward(const Tensor & grad, const Tensor & mat2, const Tensor & mat1, const Scalar & alpha) {
-  // if input was column-major, return grad as column-order for efficiency
-  if (mat1.is_sparse()) {
-    throw std::runtime_error("calculating the gradient of a sparse Tensor argument to mm is not supported.");
+// This function is used by load_derivatives.py to replace tensor.strides()
+// calls that appear in derivative formulas. This function checks if the tensor
+// is strided, and if not, produces an error telling the user how to fix the
+// issue. If requires_grad is not set for the corresponding inputs of the
+// operation, no error is thrown and an empty array is returned since there
+// will be no backward pass in this case.
+//
+// Args:
+//  self              Tensor to call .strides() on
+//  self_name         Name of `input` tensor, from derivative formula
+//  grad_vars         List of tensor inputs the derivative formula applies to
```

Comment:
Note that currently, in all cases where this function will be called, `grad_vars` just contains `self`. But since codegen is adding this call automatically, I wanted to make sure it will still work if `grad_vars` contains something other than `self`. In other words, I wanted to support a derivative formula does something like this, in case it's ever needed:
```
mat1, mat2: ... self.strides() ...
```

Reply:
I like the assert approach. It does seem better not to add functionality that isn't tested or currently needed.